### PR TITLE
Storage abstraction

### PR DESCRIPTION
**Pull request by sparty02**
@thejameskyle @hemanth  Looking for another pair of :eyes:   This is a simple abstraction to separate concerns from the REPL (not having to know as much about the storage implementation/etc.

### CODE REVIEW

**Comment by hemanth** [comment]
Cool :+1: I would avoid `new` and use a `IIFE` instead.

File: scripts/repl.js [hunk 1]
```diff
@@ -94,31 +132,29 @@
...
-        try {
-          var storedState = localStorage.getItem('replState');
-          if (storedState) {
-            state = JSON.parse(storedState);  
-          }
-        } catch(e) {}
+  function REPL (storage) {
+    var state = {};
+    if (storage) {
+      _.assign(state, storage.get('replState'));
```
Comment (thejameskyle) [inline scripts/repl.js]: I think it'd be better if the REPL just created its own storage. Something like:

```js
function REPL () {
  this.storage = new StorageService();
  var state = storage.get('replState') || {};
```

File: scripts/repl.js [hunk 2]
```diff
@@ -227,17 +263,14 @@
 
   REPL.prototype.persistState = function (state) {
     UriUtils.updateQuery(state);
-    if (window.localStorage) {
-        try {
-          window.localStorage.setItem('replState', JSON.stringify(state));
-        } catch(e) {}
-    }
+    storage.set('replState', state);
```
Comment (thejameskyle) [inline scripts/repl.js]: This is currently depending on the global `storage` object

File: scripts/repl.js (hunk 1 above)
Comment (hemanth) [inline scripts/repl.js]: You meant, `this.storage.get` or `var storage` ?

File: scripts/repl.js (hunk 2 above)
Comment (sparty02) [inline scripts/repl.js]: Agreed, this should be this.storage

File: scripts/repl.js (hunk 1 above)
Comment (sparty02) [inline scripts/repl.js]: Yep, should be this.storage

Comment (hemanth) [inline scripts/repl.js]: :+1:

**Comment by hemanth** [comment]
:+1:

**Pull request merged**